### PR TITLE
runtimeVM: ExecSync should return a recognizable error on timeout

### PR DESCRIPTION
**Pull request by littlejawa**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:
When running ExecSync with a timeout, do not report an error, but make an ExecSyncResponse showing that a timeout occurred.

This PR copies the logic found in runtime_oci.go:ExecSyncContainer(), and applies it to runtime_vm.go:ExecSyncContainer(), so that both have the same behaviour.

#### Which issue(s) this PR fixes:

This issue was found by running the integration tests with Kata as the runtime.
The test "ctr execsync" was failing because the error reported on a timeout was not the one expected.
With this PR, the test passes.

#### Does this PR introduce a user-facing change?

```release-note
none
```

### CODE REVIEW

File: internal/oci/runtime_vm.go [hunk 1]
```diff
@@ -453,7 +462,8 @@ func (r *runtimeVM) execContainerCommon(ctx context.Context, c *Container, cmd [
 			return -1, killErr
 		}
 		<-execCh
-		return -1, errors.Errorf("ExecSyncContainer timeout (%v)", timeoutDuration)
+		// do not make an error for timeout: report it with a specific error code
+		return -2, nil
```
Comment (fidencio) [inline internal/oci/runtime_vm.go:466]: Please, define an error instead of just using "-2",

**Review by fidencio — COMMENTED** [review]
@littlejawa, the patch itself looks good, but I'd like to see a test for this (could be here, could be as part of the kata-containers' bats).

Tomorrow I'll try to exercise this code locally and finally approve (or ask for chances in) this PR.

Thanks for working on this, @littlejawa!

**Comment by fidencio** [comment]
> @littlejawa, the patch itself looks good, but I'd like to see a test for this (could be here, could be as part of the kata-containers' bats).

Oh, I missed the fact this is fixing an issue found by thr ctr bats tests, sorry.
The comment about adding a new error still stands, though. :-)

File: internal/oci/runtime_vm.go (hunk 1 above)
Comment (littlejawa) [reply internal/oci/runtime_vm.go]: Done. I'm keeping it as a separate commit for now, just tell me if it's ok, and then I can squash if needed.

File: internal/oci/runtime_vm.go [hunk 2]
```diff
@@ -318,6 +325,14 @@ func (r *runtimeVM) ExecSyncContainer(ctx context.Context, c *Container, command
 		return nil, errors.Wrap(err, "ExecSyncContainer failed")
 	}
 
+	// if the execution stopped because of the timeout, report it as such
+	if exitCode == execTimeout {
+		return &types.ExecSyncResponse{
+			Stderr:   []byte(conmonconfig.TimedOutMessage),
+			ExitCode: -1,
```
Comment (fidencio) [inline internal/oci/runtime_vm.go:332]: Is this one not changed on purpose?

File: internal/oci/runtime_vm.go (hunk 1 above)
Comment (fidencio) [reply internal/oci/runtime_vm.go]: I think I prefer this in two separate commits, thanks @littlejawa!

File: internal/oci/runtime_vm.go (hunk 2 above)
Comment (littlejawa) [reply internal/oci/runtime_vm.go:332]: Yes. According to the [equivalent code un runtime_oci.go](https://github.com/cri-o/cri-o/blob/master/internal/oci/runtime_oci.go#L369), this is what the caller expects.

**Comment by fidencio** [comment]
LGTM, thanks @littlejawa!

@haircommander, @saschagrunert, may I bother you for your input here as well?

**Comment by haircommander** [comment]
I'll let @saschagrunert give the approval :)

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Pull request merged**